alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt

```diff
@@ -69,47 +71,83 @@ class DocLevelMonitorQueries(private val client: AdminClient, private val cluste
         val queries: List<DocLevelQuery> = docLevelMonitorInput.queries
 
         val clusterState = clusterService.state()
-        if (clusterState.routingTable.hasIndex(index)) {
-            val indexMetadata = clusterState.metadata.index(index)
 
-            if (indexMetadata.mapping() != null) {
-                val properties = ((indexMetadata.mapping()?.sourceAsMap?.get("properties")) as Map<String, Map<String, Any>>)
-                val updatedProperties = properties.entries.associate { "${it.key}_$monitorId" to it.value }.toMutableMap()
-
-                val updateMappingRequest = PutMappingRequest(ScheduledJob.DOC_LEVEL_QUERIES_INDEX)
-                updateMappingRequest.source(mapOf<String, Any>("properties" to updatedProperties))
-
-                queryClient.admin().indices().putMapping(
-                    updateMappingRequest,
-                    object : ActionListener<AcknowledgedResponse> {
-                        override fun onResponse(response: AcknowledgedResponse) {
-                            log.info("Percolation index ${ScheduledJob.DOC_LEVEL_QUERIES_INDEX} updated with new mappings")
-
-                            val request = BulkRequest().setRefreshPolicy(refreshPolicy).timeout(indexTimeout)
-
-                            queries.forEach {
-                                var query = it.query
-
-                                properties.forEach { prop ->
-                                    query = query.replace("${prop.key}:", "${prop.key}_$monitorId:")
-                                }
-                                val indexRequest = IndexRequest(ScheduledJob.DOC_LEVEL_QUERIES_INDEX)
-                                    .id(it.id + "_$monitorId")
-                                    .source(mapOf("query" to mapOf("query_string" to mapOf("query" to query)), "monitor_id" to monitorId))
-                                request.add(indexRequest)
+        val getAliasesRequest = GetAliasesRequest(index)
+        queryClient.admin().indices().getAliases(
+            getAliasesRequest,
+            object : ActionListener<GetAliasesResponse> {
+                override fun onResponse(getAliasesResponse: GetAliasesResponse?) {
+                    val aliasIndices = getAliasesResponse?.aliases?.keys()?.map { it.value }
+                    val isAlias = aliasIndices != null && aliasIndices.isNotEmpty()
+                    val indices = if (isAlias) getAliasesResponse?.aliases?.keys()?.map { it.value } else listOf(index)
```

Comment:
Minor question; would it make sense to refactor this to the following since the response has already been parsed to create `aliasIndices`?
```
val indices = if (isAlias) aliasIndices else listOf(index)
```

Reply:
Nice catch